src/prop/sat_proof_manager.cpp

```diff
@@ -354,20 +354,32 @@ void SatProofManager::explainLit(
   }
   // pedantically check that the negation of the literal to explain *does not*
   // occur in the reason, otherwise we will loop forever
+#ifdef CVC4_ASSERTIONS
   for (unsigned i = 0; i < size; ++i)
   {
     AlwaysAssert(~MinisatSatSolver::toSatLiteral(reason[i]) != lit)
         << "cyclic justification\n";
   }
+#endif
   // add the reason clause first
   std::vector<Node> children{getClauseNode(reason)}, args;
   // save in the premises
   premises.insert(children.back());
+  // Since explainLit calls can reallocate memory in the
+  // SAT solver, we directly get the literals we need to explain so we no
+  // longer depend on the reference to reason
+  std::vector<Node> toExplain{children.back().begin(), children.back().end()};
   NodeManager* nm = NodeManager::currentNM();
   Trace("sat-proof") << push;
   for (unsigned i = 0; i < size; ++i)
   {
-    SatLiteral curr_lit = MinisatSatSolver::toSatLiteral(reason[i]);
+    // pedantically make sure that the reason stays the same
```

Comment:
What part of the code is making sure the reason stays the same?  Its confusing to have this before the assertions block, makes it seem like this check is only done when assertions are enabled, this isnt the case though?

Reply:
I should move it into the assertion block. The expectation is that the reason does not change. I just wanted to be extra careful.